pkg/validator/cluster.go

```diff
@@ -408,6 +409,14 @@ func validateGeneralClusterChanges(
 				prevCrApp.Spec.Version,
 			)
 			valErrors = append(valErrors, appModifiedMsg)
+			return valErrors
+		}
+
+		// Also, check the new app differs from the current one in the role images and cardinality only
```

Comment:
Let's change this comment to just say "check that the new app differs only in allowed ways", or something like that, so that if we change the semantics of validateAppSpecChanges in the future we won't have to come back here to fix the comment.

Reply:
Changed